src/ore/src/channel.rs

```diff
@@ -94,6 +98,100 @@ impl<T: Send> ReceiverExt<T> for tokio::sync::mpsc::UnboundedReceiver<T> {
     }
 }
 
+/// A trait describing a metric that can be used with an `instrumented_unbounded_channel`.
+pub trait InstrumentedChannelMetric {
+    /// Bump the metric, increasing the count of operators (send or receives) that occurred.
+    fn bump(&self);
+}
+
+impl<'a, P, L> InstrumentedChannelMetric for crate::metrics::DeleteOnDropCounter<'a, P, L>
+where
+    P: Atomic,
+    L: PromLabelsExt<'a>,
+{
+    fn bump(&self) {
+        self.inc()
+    }
+}
+
+/// A wrapper around tokio's mpsc unbounded channels that connects
+/// metrics that are incremented when sends or receives happen.
+pub fn instrumented_unbounded_channel<T, M>(
+    sender_metric: M,
+    receiver_metric: M,
+) -> (
+    InstrumentedUnboundedSender<T, M>,
+    InstrumentedUnboundedReceiver<T, M>,
+)
+where
+    M: InstrumentedChannelMetric,
+{
+    let (tx, rx) = unbounded_channel();
+
+    (
+        InstrumentedUnboundedSender {
+            tx,
+            metric: sender_metric,
+        },
+        InstrumentedUnboundedReceiver {
+            rx,
+            metric: receiver_metric,
+        },
+    )
+}
```

Comment:
Maybe this would be too many traits, but what do you think about an `InstrumentedChannel` trait so someone could do
```
let (tx, rx) = mpsc::unbounded_channel().instrumented(...);
```
this approach vs `mz_ore::channel::instrumented_unbounded_channel(...)` really doesn't matter to me, just wanted to throw it out there.

Reply:
@ParkMyCar that would match the wallfuture stuff, but I don't really want to write more traits for a slight ergo win!

Reply:
Fine by me!